graphql-test/src/main/java/org/jahia/test/graphql/GraphQLPublicationTest.java

```diff
@@ -296,4 +299,81 @@ private JCRSiteNode createTestSite(String name, String template, Set<String> lan
+            JSONObject property = query.getJSONObject("data").getJSONObject("jcr")
+                    .getJSONObject("nodeByPath").getJSONObject("property");
+            logger.info("Path: /sites/{}/testList\nProperty: j:published\nValue: {}", siteName, property.getBoolean("value"));
+
+            final JCRNodeWrapper node = session.getNode("/sites/" + siteName);
+            Assert.assertTrue("Site" + siteName + " should have node testList", node.hasNode("testList"));
+            logger.info("Site: {} has node 'testList': {}", siteName, node.hasNode("testList"));
+            if (!node.hasNode("testList")) {
+                query = executeQuery(""
+                        + "query {"
+                        + "  jcr(workspace: LIVE) {"
+                        + "     nodeByPath(path: \"/sites/"+ siteName + "\") {"
+                        + "         children(names: [\"testList\"]) {"
+                        + "            nodes {"
+                        + "              name"
+                        + "            }"
+                        + "         }"
+                        +"      }"
+                        +"   }"
+                        +"}"
+                );
+                JSONArray nodes = query.getJSONObject("data").getJSONObject("jcr")
+                        .getJSONObject("nodeByPath").getJSONObject("children").getJSONArray("nodes");
+                for (int index = 0; index < nodes.length(); index++ ) {
+                    logger.info("Node names for /sites/{}: {}", siteName, nodes.getString(index));
+                }
+            }
+        } else {
+            logger.error("/sites/{} not found. It may have been deleted", siteName);
+        }
+    }
+
+    private void waitForPublicationToFinish() throws SchedulerException {
+
+        // Method #1
+        long startedWaitingAt = System.currentTimeMillis();
+
+        // Wait until the node is published via a background job.
+        while(Arrays.stream(schedulerService.getScheduler().getTriggerNames(SchedulerService.INSTANT_TRIGGER_GROUP))
+                .anyMatch(t -> t.contains("Publication"))) {
+            if (System.currentTimeMillis() - startedWaitingAt > TIMEOUT_WAITING_FOR_PUBLICATION) {
+                Assert.fail("Timeout waiting for node to be published");
+            }
+        }
+
+
+        // Method #2
+        /*
+        with().pollInterval(ONE_SECOND).await().atMost(TIMEOUT_WAITING_FOR_PUBLICATION, MILLISECONDS)
+                .ignoreExceptions()
+                .until(new Callable<Boolean>() {
+                    @Override public Boolean call() throws Exception {
+                        final String[] triggerNames = schedulerService.getScheduler()
+                                .getTriggerNames(SchedulerService.INSTANT_TRIGGER_GROUP);
+                        return Arrays.stream(triggerNames).noneMatch(t -> t.contains("Publication"));
+                    }
+                }, equalTo(true));
+         */
+    }
```

Comment:
To make awaitility work you probably need to add the library to the pom.xml so it gets packaged with the module